### DISCUSSION TITLE
Make text objects work for next occurence

### DISCUSSION THREAD

**Discussion by maxime-aknin** [JetBrains/ideavim#294] [General]
A neat feature of vim is to allow the use of text objects even when the cursor is not at the right position.
E.g One can use ci" and if the cursor is not inside quotes it will find the next quotes and do the magic (and sometimes it even works if not on the same line, depending on which vim/neovim version it is I guess).

Here's a video showcasing the concept: https://youtu.be/E7NBhSsZouc

That would be very nice to have the same feature in IdeaVim !

Thanks

**Comment by AlexPl292** [comment]
Hi! Could you please show an example where this doesn't work? Because a simple example works pretty well for me. 
(because of some reason keys were not captured on this gif, but here I type `ci"`.)
![2021-04-20 11 23 25](https://user-images.githubusercontent.com/4203721/115363213-ee651600-a1ca-11eb-9987-e7354b82a32a.gif)

**Reply by maxime-aknin** [reply]
Hi !

Sorry my mistake, indeed it seems to work on a line basis (I think I just wasn't typing fast enough when I tested it 🤦‍♂️). 
**EDIT:** in fact it's not working for all characters, see my other answer below.

However I can confirm it doesn't work on a different line (which is a nice feature, I think it might be nvim specific though), here's a capture I made showing it: 
![ideavim](https://user-images.githubusercontent.com/8659993/115407643-7e698680-a1f0-11eb-8097-38fb81764d0c.gif)

Thank you !

**Reply by maxime-aknin** [reply]
Hi again,

Sorry for the double response, but I found out what wasn't working on a line basis the first time, in fact it's not working with the parenthesis and the brackets (here's another capture below), that's what misguided me in the first place I assumed the quotes did not work either !
![ideavim2](https://user-images.githubusercontent.com/8659993/115410120-a8bc4380-a1f2-11eb-9625-9fe70b6fd6ed.gif)
 
Here's [my ideavimrc](https://github.com/maxime-aknin/.vim/blob/main/.ideavimrc) in case you need to see what settings I have

**Comment by AlexPl292** [comment]
Hmm, you know, this is a bit confusing, but I've just played with vim and nvim for some time and I can't say that any of the described behaviours work for me (`ci"` from different line and `cib` on the same line). I also tried to use vim with your .vimrc config (from here https://github.com/maxime-aknin/.vim/blob/main/.vimrc) and I still cannot reproduce it. 

Could you maybe share your nvim version (with `nvim --version`) and try to check this without loading your configurations (for vim it's `vim -u NONE`, so I guess that the same works with `nvim`)?

**Reply by maxime-aknin** [reply]
Hi Alex,

Yes sorry my mistake again, indeed after a little bit of digging I found out this behaviour is in fact coming from a [plugin](https://github.com/wellle/targets.vim#in-pair) and not from vim / neovim itself, I though this plugin only added new text objects but in fact it seems to also add this feature.

So should I rephrase my proposition it would be to maybe add this plugins features as an ideavim plugin (like surround and others) ? I know it must not be an easy task so it's just a suggestion. I would gladly help and contribute, but I'd have to invest some time in learning kotlin or java beforehand (sadly I don't have any experience with these languages 😅)

Thank you

**Reply by AlexPl292** [reply]
Oh, glad to see that you've found the reason :D
Okay, I've checked this plugin and it looks.. pretty large :) I don't think it is too complicated, but it would definitely take some time to restore this functionality. If you like, you can definitely try to contribute (maybe create some basis of this plugin). You can start here: https://github.com/JetBrains/ideavim/blob/master/CONTRIBUTING.md. We already have some plugins that create new text objects, so you can check how they are implemented. 

I've found an issue for this: https://youtrack.jetbrains.com/issue/VIM-1960
